### DISCUSSION TITLE
use of venv versus using the default python on Mac/Windows

### DISCUSSION THREAD

**Discussion by codetricity** [8bitUHM/theta360-automotive-guide#43] [General]
Are people using [venv](https://docs.python.org/3/library/venv.html) or installing the Python packages directly into their system?  

Personally, I use venv, but I noticed that the README didn't specify this as part of the install steps.    For most people, maybe there isn't a significant downside to installing the django packages directly into the system python area?

If people are on Windows, are they using [WSL](https://learn.microsoft.com/en-us/windows/wsl/) or the Windows shell (like Powershell)?

On Windows 11, I have been using WSL with venv, but I'm wondering if people can just install Python on Windows and everything works?

**Comment by jcasman** [comment]
I use venv. I seem to have trouble without using it.

**Comment by codetricity** [comment]
There is an existing version of python that is installed with the macOS in most cases.  There may be other versions that people installed with homebrew.

You can control the version of Python installed on the mac with pyenv

https://youtu.be/1F2IK7CU76U?feature=shared

Once you have python installed with the version want, you can then use venv to install Python modules specific to each project.

If you don't use venv, you may get a conflict with module version.  For example, if you have an old site that requires older modules, like django-suit or a specific version of django rest framework, then you may run into problems working on older sites and newer sites.  

With venv, you can control the python version and module version per project.

However, it does require an extra step, which may not be needed if the person only works on newer Python projects.  Over time, problems with the workflow may arise.

BTW, the most recent version of Python is now 3.13.0.   You don't necessarily need to update Python to the latest stable release unless a newer package requires it, which is unlikely.  However, if you use pyenv, it is easy to use the latest stable release.

https://www.python.org/downloads/

```bash
python --version
Python 3.13.0
```
